test/cpp/common/alarm_test.cc

```diff
@@ -438,6 +439,34 @@ TEST(AlarmTest, UnsetDestruction) {
   Alarm alarm;
 }
 
+TEST(AlarmTest, CallbackSetInCallback) {
+  auto c = std::make_shared<Completion>();
+  struct GuardedAlarm {
+    grpc::internal::Mutex mu;
+    Alarm alarm ABSL_GUARDED_BY(mu);
+  } ga;
+  {
+    grpc::internal::MutexLock l(&ga.mu);
+    ga.alarm.Set(std::chrono::system_clock::now() + std::chrono::seconds(1),
+                 [&](bool ok) {
+                   EXPECT_TRUE(ok);
+                   grpc::internal::MutexLock l(&ga.mu);
+                   ga.alarm.Set(std::chrono::system_clock::now() +
+                                    std::chrono::seconds(1),
+                                [&](bool ok) {
+                                  EXPECT_TRUE(ok);
+                                  std::lock_guard<std::mutex> l(c->mu);
+                                  c->completed = true;
+                                  c->cv.notify_one();
+                                });
+                 });
+  }
+  std::unique_lock<std::mutex> l(c->mu);
+  EXPECT_TRUE(c->cv.wait_until(
```

Comment:
`cv.wait` should suffice, the test has a timeout already.

Reply:
done